rfcs/0131-sri-hashes.md

```diff
@@ -0,0 +1,50 @@
+---
+feature: sri-hashes
+start-date: 2022-07-25
+author: Winter
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Use SRI hashes and the `hash` parameter for fetchers.
+
+# Motivation
+[motivation]: #motivation
+
+Currently in Nixpkgs, the hash formats that are used in parameters to fetchers range from SRI and base-32 (the most common ones) to hex. In addition to this, packages use the algorithm-specific arguments (`sha256`, `sha512`, etc.) or the generic `hash` argument at random. This creates inconsistencies that we'd rather not have.
+
+# Detailed design
+[design]: #detailed-design
+
+1. Require all new packages to use SRI hashes + `hash` when possible
+2. Do a treewide migration for the majority of cases (single fetcher in `src`, using a fetcher that supports `hash` and SRI hashes (`fetchFromGitHub`, `fetchurl`, etc.)) to `hash` + SRI hashes
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+None at this time.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Do nothing: further inconsistencies will continue to be introduced
```

Comment:
I think there are a number of alternatives.

Different rules can be considered. For each of the following formats it could be decided to allow/disallow/preferred:

* `hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";`
* `sha256 = "0000000000000000000000000000000000000000000000000000000000000000";`
* `sha256 = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";`
* `sha256 = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";`

I've gathered there are a number of ways these rules can be 'enforced':

* Throw evaluation errors upon not conforming to rule.
* Have a it be a potential future linter rule.
* Have PR reviewers nudge contributors to follow the rule
* Do nothing: keep things like the are right now. Reviewers will suggest their own opinion in PRs if they'd like.
* Allow any format: do not allow PR reviewers to suggest changing the format in PRs to their preferred format. If the format is allowed, just let it be. This avoids PRs being always potentially invalid due to reviewers having different opinions.

Reply:
> Allow anything: Do not allow PR reviewers to suggest changing the format in PRs. If the format is allowed, just let it be. Avoid conflicting reviewer opinions.

I'm mentioning this option, as at the moment PRs are being reviewed where the reviewers suggest their own 'opinion' on the matter; even though there is no real rule. One outcome of this RFC could be to just let it be.